README.md

```diff
@@ -2,115 +2,119 @@
+- `WadZeroable::zero()`/`RayZeroable::zero()`: Returns the zero value for `Wad` and `Ray` respectively
+- `is_zero()`: Returns true if the `Wad` or `Ray` value is zero
+- `is_non_zero()` Returns true if the `Wad` or `Ray` value is not zero
 
-### Aggregation
+#### Bound values
 
-These functions operate on `wad` only.
+Both `Wad` and `Ray` implement the `BoundedInt` trait.
+- `BoundedWad::max()`: Returns the maximum `Wad` value of 2<sup>128</sup> - 1
+- `BoundedWad::min()`: Returns the minimum `Wad` value of 0
+- `BoundedRay::max()`: Returns the maximum `Ray` value of 2<sup>128</sup> - 1
+- `BoundedRay::min()`: Returns the minimum `Ray` value of 0
 
-- `floor(n)` - Round a value down to the nearest `wad`
-- `ceil(n)` - Round a value up to the nearest `wad`
+### Comparisons
+
+Both `Wad` and `Ray` implement the `PartialEq` and `PartialOrd` traits.
+
+Comparison can be performed as follows:
+- `a == b`
+- `a != b`
+- `a > b`
+- `a >= b`
+- `a < b`
+- `a <= b`
+where both `a` and `b` are of the same type.
 
 These functions operate on any `ufelt` values in the range [0, 2<sup>125</sup>], including `wad` and `ray`.
 - `unsigned_min(a, b)` - Returns the smaller of two values
 - `unsigned_max(a, b)` - Returns the larger of two values
 
 ### Conversion
 
-- `to_uint(n)`: Convert a felt value to `Uint256`
-- `from_uint(n)`: Convert a `Uint256` value to felt with a bounds check that the result is within the range [-2<sup>125</sup>, 2<sup>125</sup>]
-- `to_wad(n)`: Multiply `n` by 10<sup>18</sup> with a bounds check that the result is within the range [-2<sup>125</sup>, 2<sup>125</sup>]
-- `wad_to_ray(n)`: Multiply `n` by 10<sup>9</sup> with a bounds check that the result is within the range [-2<sup>125</sup>, 2<sup>125</sup>]
-- `ray_to_wad(n)`: Divide `n` by 10<sup>9</sup>
-
-### Bounds Check
+The following conversions are supported by implementing the `Into` trait:
+- `Ray` -> `Wad`: Divide the `Ray` value by 10<sup>9</sup> and return a `Wad`
+- `Wad` -> `felt252`: Convert a `Wad` to a `felt252`
+- `Wad` -> `u256`: Convert a `Wad` to a `u256`
 
-These functions operate on both `wad` and `ray`.
-
-- `assert_result_valid(n)`: Raises an error if `n` is not in the range [-2<sup>125</sup>, 2<sup>125</sup>]
-- `assert_result_valid_unsigned(n)`: Raises an error if `n` is not in the range [0, 2<sup>125</sup>]
+The following conversions are supported by implementing the `TryInto` trait:
+- `Wad` -> `Option::<Ray>`: Multiply the `Wad` value by 10<sup>9</sup> and return `Option::Some<Ray>` if there is no overflow or `Option::None` otherwise.
+- `u256` -> `Option::<Wad>`: Returns `Option::Some<Wad>` if the value is within bounds of `u128` or `Option::None` otherwise.
 
 ## Usage
 
-To use this library, include a copy of `wad_ray.cairo` in your project, and import the library into your Cairo contracts.
+To use this library, include a copy of `wadray.cairo` in your project, and import the library into your Cairo contracts.
```

Comment:
The lib should be a Scarb package. Installation will then be just adding one line into `Scarb.toml` [dependencies] (and eventually, once Scarb has a package registry, even just `scarb add cairo-wadray`).

Reply:
Done! I will verify that the package installation works once this is merged.